### PR TITLE
add missing CFLAG

### PR DESCRIPTION
**Pull request by avoidr**
#412

### CODE REVIEW

**Comment by avoidr** [comment]
#412

File: ffmpeg.c [hunk 1]
```diff
@@ -578,8 +579,8 @@ static int ffmpeg_read_comments(struct input_plugin_data *ip_data, struct keyval
 	GROWING_KEYVALS(c);
 
 	ffmpeg_read_metadata(&c, ic->metadata);
-	for (unsigned i = 0; i < ic->nb_streams; i++) {
-		ffmpeg_read_metadata(&c, ic->streams[i]->metadata);
+	for (j = 0; j < ic->nb_streams; j++) {
+		ffmpeg_read_metadata(&c, ic->streams[j]->metadata);
```
Comment (flyingmutant) [inline ffmpeg.c]: Sorry, what exactly does this change do?

**Comment by nefthy** [comment]
the problem does not lie in ffmpeg.c. The check_ffmpeg does not pass the correct CFLAGS and the compilation fails.

File: ffmpeg.c (hunk 1 above)
Comment (avoidr) [inline ffmpeg.c]: Was supposed to fix the compilation error related to -std, by not mixing code and declarations. Both ugly fixes, but I didn't know how to fix it properly. Thought you'd probably know better. Consider this a documentation of the bug.

**Comment by avoidr** [comment]
Simply adding the -std flag to $cflags in check_ffmpeg is ugly, right?

**Comment by nefthy** [comment]
that fixes the issue, but i'm no expert in configure magic...

diff --git i/configure w/configure
index 54c59b0..415d533 100755
--- i/configure
+++ w/configure
@@ -519,6 +519,7 @@ var_default exampledir "${datadir}/doc/cmus/examples"
 check check_cc
 check check_host_cc
 check check_cflags
+CFLAGS="${CFLAGS} -DHAVE_CONFIG"
 check check_cc_depgen
 check check_endianness
 check check_compat
@@ -593,7 +594,6 @@ config_header config/samplerate.h HAVE_SAMPLERATE
 config_header config/xmalloc.h HAVE_STRDUP HAVE_STRNDUP
 config_header config/cue.h CONFIG_CUE CONFIG_CUE2
 
-CFLAGS="${CFLAGS} -DHAVE_CONFIG"
 
 makefile_vars bindir datadir libdir mandir exampledir
 makefile_vars CONFIG_CDIO CONFIG_FLAC CONFIG_MAD CONFIG_MIKMOD CONFIG_MODPLUG CONFIG_MPC CONFIG_VORBIS CONFIG_OPUS CONFIG_WAVPACK CONFIG_WAV CONFIG_MP4 CONFIG_AAC CONFIG_FFMPEG CONFIG_CUE CONFIG_CUE2 CONFIG_VTX

**Comment by nefthy** [comment]
@flyingmutant what is your take on this?

**Comment by flyingmutant** [comment]
@nefthy @avoidr sorry for delayed reply, I've been (and am) super-busy for the past couple of months.

The better way is fixing `configure`, of course – modifying `cflags` [here](https://github.com/cmus/cmus/blob/master/configure#L406).

**Comment by avoidr** [comment]
Like this now? You can also fix that yourself, I don't mind.

**Comment by flyingmutant** [comment]
Congrats on becoming a git god, that'll definitely help in the future :-)

**Pull request merged**